git_notes.md

```diff
@@ -0,0 +1,34 @@
+# Git Notes research
+
+The aim of this research is to provide
+a feedback around [git notes](https://git-scm.com/docs/git-notes) command.
+
+[Examples](./examples) directory contains [git_notes.sh](./examples/git_notes.sh).
+The command adds number of your specific commits into your `git` repository.
+
+## What git_notes.sh does?
+
+* Add N-notes into your `git repository`.
+  * The note added into git is:
+
+```Testing notes ${cnt}\npackit_test=yes"```
+* Show all notes added by previous command
+* Show specific (3th) note from git.
+
+## Can be used git notes as a database?
+
+`git notes` are basically one liners. We can use a separator and for
+some things it can be used as a "database".
+
+## Can be used git notes for storing states?
+
+Yes, definitely. We can use `git notes` for storing state.
+But, first of all, we have to read note from specific commit and then
```

Comment:
can one commit have multiple notes?

Reply:
We can use -m option multi-times, but basically one note pre one commit. Multiple -m options are concatenated as separated paragraphs. See bash script.